src/mlpack/core/math/lin_alg.cpp

```diff
@@ -77,15 +77,19 @@ void mlpack::math::WhitenUsingSVD(const arma::mat& x,
  * Whitens a matrix using the eigendecomposition of the covariance matrix.
  * Whitening means the covariance matrix of the result is the identity matrix.
  */
-void mlpack::math::WhitenUsingEig(const arma::mat& x,
+bool mlpack::math::WhitenUsingEig(const arma::mat& x,
                                   arma::mat& xWhitened,
                                   arma::mat& whiteningMatrix)
 {
   arma::mat diag, eigenvectors;
   arma::vec eigenvalues;
 
   // Get eigenvectors of covariance of input matrix.
-  eig_sym(eigenvalues, eigenvectors, mlpack::math::ColumnCovariance(x));
+  if (!eig_sym(eigenvalues, eigenvectors, mlpack::math::ColumnCovariance(x)))
+  {
+      Log::Fatal << "Failed Whitening" << std::endl;
```

Comment:
`Log::Fatal` implies the program would crash, so actually I would say just return `false` here and no error message.  (We should update the documentation to point that out.)  Then, we just need to modify whatever calls `WhitenUsingEig()` in order to handle the return code. :+1:

Reply:
Consider a Case where `A()` is calling `B()` and b is calling `WhitenUsingEig()`, and if we are returning false and handling that in `B()` Like you suggested and then we should handle in `A()` too, right. 

I guess we should use Log::Warn, just and avoid returning anything instead of handling every call to `WhitenUsingEig()`

Reply:
Actually, you know, I found that there is no code that currently uses `WhitenUsingEig()`!  (Maybe code was using it at some point but is no longer.)  So, we could either remove it or just change the return code.  Up to you which you want to do.